### PR TITLE
Additional TDE configuration for Patroni pg_rewind

### PR DESCRIPTION
**Pull request by tjmoore4**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [ ] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?
   - [ ] Have you added automated tests?

**Type of Changes:**

 
 - [x] New feature
 - [ ] Bug fix
 - [ ] Documentation
 - [ ] Testing enhancement
 - [ ] Other

**What is the new behavior (if this is a feature change)?**
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

Provide a wrapper script to allow Patroni to invoke pg_rewind as required for TDE and update configuration to enable pg_rewind in Patroni for all versions > 10.

**Other Information**:
Issue: PGO-785

### CODE REVIEW

**Review by ValClarkson — APPROVED** [review]

File: internal/postgres/reconcile_test.go [hunk 1]
```diff
@@ -251,6 +251,7 @@ initContainers:
     halt "$(permissions "${pgbrLog_directory}" ||:)"
     install -D --mode=0600 -t "/tmp/replication" "/pgconf/tls/replication"/{tls.crt,tls.key,ca.crt}
 
+
```
Comment (tony-landreth) [inline internal/postgres/reconcile_test.go:254]: Accidental check-in?

File: internal/patroni/config.go [hunk 2]
```diff
@@ -283,8 +292,7 @@ func DynamicConfiguration(
 	// Additionally, Patroni does not currently provide an easy way to set the
 	// "encryption_key_command" value for `pg_rewind`, so if TDE is enabled,
 	// `use_pg_rewind` is set to false.
```
Comment (tony-landreth) [inline internal/patroni/config.go:294]: Should this comment be updated?

File: internal/postgres/reconcile_test.go (hunk 1 above)
Comment (tjmoore4) [reply internal/postgres/reconcile_test.go:254]: No, that's actually required for the test. When TDE is not enabled, an extra newline is added due to the way the change is made. It follows a similar pattern as the existing [tablespace command](https://github.com/CrunchyData/postgres-operator/blob/master/internal/postgres/config.go#L335).

Comment (tony-landreth) [reply internal/postgres/reconcile_test.go:254]: Gotcha. Thanks!

File: internal/postgres/config.go [hunk 3]
```diff
@@ -246,6 +247,18 @@ func startupCommand(
 		}
 	}
 
+	pg_rewind_override := ""
+	if config.FetchKeyCommand(&cluster.Spec) != "" {
+		// Quoting "EOF" disables parameter substitution during write.
+		// - https://tldp.org/LDP/abs/html/here-docs.html#EX71C
+		pg_rewind_override = `cat << "EOF" > /tmp/pg_rewind_tde.sh
+#!/bin/sh
+pg_rewind -K "$(postgres -C encryption_key_command)" "$@"
```
Comment (tony-landreth) [inline internal/postgres/config.go:256]: Does the `"$@"` get used?

Comment (tjmoore4) [reply internal/postgres/config.go:256]: Yes, that allows the other required flags to be passed to `pg_rewind` when it's invoked by Patroni.

Comment (tony-landreth) [reply internal/postgres/config.go:256]: That makes sense. Thanks!

**Review by ValClarkson — APPROVED** [review]

**Pull request merged**